src/test/lib/logical_query_plan/aggregate_node_test.cpp

```diff
@@ -94,4 +95,105 @@ TEST_F(AggregateNodeTest, Copy) {
+  {
+    EXPECT_EQ(agg_node_a->unique_constraints()->size(), 1);
+    const auto unique_constraint = *agg_node_a->unique_constraints()->cbegin();
+    EXPECT_EQ(unique_constraint.expressions.size(), 1);
+    EXPECT_TRUE(unique_constraint.expressions.contains(_a));
+  }
+  {
+    EXPECT_EQ(agg_node_b->unique_constraints()->size(), 1);
+    const auto unique_constraint = *agg_node_b->unique_constraints()->cbegin();
+    EXPECT_EQ(unique_constraint.expressions.size(), 2);
+    EXPECT_TRUE(unique_constraint.expressions.contains(_a));
+    EXPECT_TRUE(unique_constraint.expressions.contains(_b));
+  }
+}
+
+TEST_F(AggregateNodeTest, UniqueConstraintsForwardingSimple) {
+  const auto key_constraint_b = TableKeyConstraint{{_b->original_column_id}, KeyConstraintType::UNIQUE};
+  const auto key_constraint_c = TableKeyConstraint{{_c->original_column_id}, KeyConstraintType::UNIQUE};
+  _mock_node->set_key_constraints({key_constraint_b, key_constraint_c});
+  EXPECT_EQ(_mock_node->unique_constraints()->size(), 2);
+
+  const auto aggregate_c = sum_(_c);
+  _aggregate_node = AggregateNode::make(expression_vector(_a, _b), expression_vector(aggregate_c), _mock_node);
+  const auto& unique_constraints = _aggregate_node->unique_constraints();
+
+  /**
+   * Expected behaviour:
+   *  - key_constraint_b remains valid since _b is part of the group-by columns.
+   *  - key_constraint_c, however, should be discarded because _c gets aggregated.
+   *  - Also, we should gain a new unique constraint, covering all group-by columns.
+   */
+
+  // Basic check
+  EXPECT_EQ(unique_constraints->size(), 2);
+  // In-depth check
+  EXPECT_TRUE(find_unique_constraint_by_key_constraint(key_constraint_b, unique_constraints));
+  const auto key_constraint_group_by =
+      TableKeyConstraint{{_a->original_column_id, _b->original_column_id}, KeyConstraintType::UNIQUE};
+  EXPECT_TRUE(find_unique_constraint_by_key_constraint(key_constraint_group_by, unique_constraints));
+}
+
+TEST_F(AggregateNodeTest, UniqueConstraintsForwardingAnyAggregates) {
+  const auto key_constraint_b = TableKeyConstraint{{_b->original_column_id}, KeyConstraintType::UNIQUE};
+  const auto key_constraint_c = TableKeyConstraint{{_c->original_column_id}, KeyConstraintType::UNIQUE};
+  _mock_node->set_key_constraints({key_constraint_b, key_constraint_c});
+  EXPECT_EQ(_mock_node->unique_constraints()->size(), 2);
+
+  const auto aggregate_b = any_(_b);
+  const auto aggregate_c = sum_(_c);
+  _aggregate_node = AggregateNode::make(expression_vector(_a), expression_vector(aggregate_b, aggregate_c), _mock_node);
+  const auto& unique_constraints = _aggregate_node->unique_constraints();
+
+  /**
+   * Expected behaviour:
+   *  - key_constraint_b remains valid because _b is aggregated via ANY(), a pseudo aggregate function used
+   *    by the DependentGroupByReductionRule to optimize group-bys.
+   *  - key_constraint_c should be discarded because _c is aggregated.
```

Comment:
Another possibility for the future that we could as a comment to the node: If `c` is unique, so is `MAX(c)` (but not `SUM(c)`.

I wonder if there is a comprehensive list of all of these rules.

Reply:
Good point. I added a note in AggregateNode.